Some changes

iyegoroff opened the pull request:
- Added minHeight prop;
- removed hardcoded underlineColorAndroid;
- added typescript typings;
- removed lodash dependency;
- added shrinkIfEmpty prop (if true height will be set to minHeight when props.value is falsy);

iyegoroff committed:
added minHeight & removed underlineColorAndroid

iyegoroff committed:
added index.d.ts

iyegoroff committed:
added typings

iyegoroff committed:
m

iyegoroff committed:
fix

iyegoroff committed:
lodash.omit

iyegoroff committed:
changed lodash to lodash.omit

iyegoroff committed:
removed lodash.omit

iyegoroff committed:
removed dependencies

iyegoroff committed:
shrinkIfEmpty

iyegoroff committed:
added shrinkIfEmpty

iyegoroff committed:
fix

rusfearuth commented:
@iyegoroff Hi! Could you give more information about your PR? What will it bring to exist UX?

iyegoroff commented:
Hi, @rusfearuth! I have added minHeight prop, because it didn't work for me correctly with default value = 30 when the input was empty. Removed underlineColorAndroid={'transparent'}, so users can pass this color via props. Also added shrinkIfEmpty prop to minimize the height of component, when input is cleared programmatically, e.g. when you type chat message then press 'send', the message is being sent to server and the input is being cleared - at this point component doesn't resize and onChange is not called (at least on Android), so input will change its height only if shrinkIfEmpty equals true.

rusfearuth commented:
@iyegoroff Okay. Thx for PR.

Please add passing underlineColorAndroid as prop to Component and apply my code style for `_calcHeight` and `_minHeight`. I just don't want to have mix of code style in component.

rusfearuth committed:
Stuck of height in Android

If component has more than 1 lines, it'll keep height on `value` = ''.
